lib/x11utils.pm

```diff
@@ -76,8 +76,16 @@ sub ensure_unlocked_desktop {
 
     while ($counter--) {
         my @tags = qw(displaymanager displaymanager-password-prompt generic-desktop screenlock screenlock-password authentication-required-user-settings authentication-required-modify-system guest-disabled-display oh-no-something-has-gone-wrong);
-        push(@tags, 'blackscreen') if get_var("DESKTOP") =~ /minimalx|xfce/;    # Only xscreensaver and xfce have a blackscreen as screenlock
+        push(@tags, 'blackscreen')      if get_var("DESKTOP") =~ /minimalx|xfce/;    # Only xscreensaver and xfce have a blackscreen as screenlock
+        push(@tags, 'gnome-activities') if check_var('DESKTOP', 'gnome');
         assert_screen \@tags, no_wait => 1;
+        # Starting with GNOME 40, upon login, the activities screen is open (assuming the
+        # user will want to start something. For openQA, we simply press 'esc' to close
+        # it again and really end up on the desktop
+        if (match_has_tag('gnome-activities')) {
+            send_key 'esc';
+            @tags = grep { !/gnome-activities/ } @tags;
+        }
```

Comment:
Generally ok - just that we have this probably like 3 or 4 times in our codebase already.

We might want to try to further refactor that at one point (can be after we get the tests stable again imho)